add an agenda sneak peek

luto opened the pull request:
Closes #239. Spiritual successor to #305.

## How Has This Been Tested?

By frantically clicking around. There are also some automatic tests.

## Screenshots (if appropriate):

![image](https://user-images.githubusercontent.com/2158203/34489252-e35dee16-efdb-11e7-832b-189e2ee296bc.png)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- It is ok to not check all boxes! We just want to know if we need to do any work after merging the PR. -->

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] My change is listed in the CHANGELOG.rst if appropriate.

## TODO

- [x] add an event setting to enable or disable this
- [x] fix permissions, so this can actually be viewed _before_ the schedule is public
- [x] link this on the event start page (only if enabled)
- [x] think about how do choose which talks to display
  - add a per-talk flag
  - show random, confirmed talks each time site is shown
  - show random, confirmed talks each day (using the date as seed)
- [ ] add pagination?
- [ ] allow admins the configure the text shown above the talks?
  - there are two texts: one before any talks have been added, one after. Allow them to configure both? Only display the configured one, if one is configured?
- [x] add an orga view to add talks to sneak peek easily
- [x] test that CONFIRMED talks show up
- [x] test that _only_ CONFIRMED talks show up
- [x] test that schedule is shown, if sneak enabled and schedule present
- [x] test that sneak is shown, if sneak enabled and no schedule present

luto commented:
I'll play around with this during the next days. So please don't merge yet. :) I'm happy to take a look at any reviews or comments, though!

rixx commented on src/pretalx/agenda/permissions.py:
```diff
@@ -25,7 +35,8 @@ def is_speaker_viewable(user, profile):
 
 
-rules.add_perm('agenda.view_schedule', is_agenda_visible | is_orga)
+rules.add_perm('agenda.view_schedule', (has_agenda & is_agenda_visible) | is_orga)
```
:+1:

rixx commented on src/pretalx/agenda/templates/agenda/sneakpeek.html:
```diff
@@ -0,0 +1,40 @@
+<p>
+    <b>{% trans "Welcome to our schedule sneak peek!" %}</b><br><br>
+    {% if talks %}
+    {% blocktrans %}
```
``blocktrans trimmed``, please.

rixx commented on src/pretalx/agenda/templates/agenda/sneakpeek.html:
```diff
@@ -0,0 +1,40 @@
+    <b>{% trans "Welcome to our schedule sneak peek!" %}</b><br><br>
+    {% if talks %}
+    {% blocktrans %}
+    We prepared a list of exciting talks, so you canget a feel for our
```
"canget" ;)

rixx commented on src/pretalx/agenda/templates/agenda/sneakpeek.html:
```diff
@@ -0,0 +1,40 @@
+    {% blocktrans %}
+    We prepared a list of exciting talks, so you canget a feel for our
+    conference. Please keep in mind that this is not our full schedule.
+    That will follow in due time.
```
"We will follow up with the full schedule in time, stay tuned!" maybe?

rixx commented on src/pretalx/agenda/templates/agenda/sneakpeek.html:
```diff
@@ -0,0 +1,40 @@
+    That will follow in due time.
+    {% endblocktrans %}
+    {% else %}
+    {% blocktrans %}
```
``blocktrans trimmed``, please.

rixx commented on src/pretalx/agenda/views/sneakpeek.py:
```diff
@@ -0,0 +1,16 @@
+    permission_required = 'agenda.view_sneak_peek'
+
+    def get_permission_object(self):
+        return self.request.event
```
I think it would be neat to redirect to the actual agenda once it exists and is visible. (You could do that in ``dispatch``).

luto replied:
done. :)

rixx commented on src/pretalx/orga/forms/event.py:
```diff
@@ -101,6 +101,11 @@ class EventSettingsForm(ReadOnlyFlag, I18nFormMixin, HierarkeyForm):
     )
+    show_sneak_peek = forms.BooleanField(
+        label=_('Show a sneak peek before schedule release'),
+        help_text=_('Set to publicly display a list of talks, which have the "is featured" flag enabled.'),
```
"Set to display a public list of talks prior to a schedule release. You can configure the talks in the … area."

luto commented:
rebased, added a little js-driven checkbox for each talk in the orga submission list :) depending on which additional features you'd like to see, I'd consider this done for now (assuming the tests pass).

rixx commented:
Thank you. :space_invader:
